lncfg/db.go

```diff
@@ -91,46 +95,58 @@ type DatabaseBackends struct {
 	// contains the critical channel state related data.
 	ChanStateDB kvdb.Backend
 
+	// MacaroonDB points to a database backend that stores the macaroon root
+	// keys.
+	MacaroonDB kvdb.Backend
+
 	// Replicated indicates whether the database backends are remote, data
 	// replicated instances or local bbolt backed databases.
 	Replicated bool
 }
 
 // GetBackends returns a set of kvdb.Backends as set in the DB config.
-func (db *DB) GetBackends(ctx context.Context, dbPath string) (
+func (db *DB) GetBackends(ctx context.Context, chanDBPath string,
+	makeMacaroonBoltDB boltBackendCreator) (
 	*DatabaseBackends, error) {
 
 	if db.Backend == EtcdBackend {
 		etcdBackend, err := kvdb.Open(
 			kvdb.EtcdBackendName, ctx, db.Etcd,
 		)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error opening etcd DB: %v", err)
 		}
 
 		return &DatabaseBackends{
 			GraphDB:     etcdBackend,
 			ChanStateDB: etcdBackend,
+			MacaroonDB:  etcdBackend,
```

Comment:
The nice thing about the `etcd` backend is that we could actually instantiate multiple backends with different namespaces too. This'd also allow higher throughput since all these can work in parallel (each one having it's own commit queue). Not a scope of this PR though, can be done later as an optimization.

Reply:
Namespace separation is not required, it's just a nice thing to have and would allow separate snapshotting and backup.

Reply:
Crazy idea, we could also separate things into different etcd clusters for even better scalability :)

Reply:
Yes, the namespacing is in the last commit to make the changes very contained. This will allow us to separate things out even further in the future, correct. Maybe not only different clusters but also different backend technologies? E.g. something fast for the graph and something more on the reliable side for the channel state?

Reply:
Exactly